### PR TITLE
Added coverage test 0451

### PR DESCRIPTION
**Pull request by Vihanakangas**

### CODE REVIEW

File: coverage/PENDING_REVIEW/intel/0451-COVERAGE-nir-loop-analyze-834-937-995/cov-texel-double-negation.amber [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+
+layout(location = 0) out vec4 _GLF_color;
+
+void main()
+{
+ _GLF_color = vec4(
+ gl_FragCoord.x * (1.0 / 256.0),
+ (int(gl_FragCoord.x) ^ int(gl_FragCoord.y)) * (1.0 / 256.0),
+ gl_FragCoord.y * (1.0 / 256.0),
+ 1.0);
```
Comment (afd) [inline coverage/PENDING_REVIEW/intel/0451-COVERAGE-nir-loop-analyze-834-937-995/cov-texel-double-negation.amber:38]: I don't understand why `gl_FragCoord.x * (1.0 / 256.0)` is guaranteed to come out as 1.0 for all fragments and `gl_FragCoord.y * (1.0 / 256.0)` guaranteed to come out as 0.0 for all fragments. Surely if there are two fragments whose x and y coordinates are swapped, the result of one of these expressions for the first fragment would be the same as the result of the other expression for the second fragment.

I also find the ^ logic hard to follow and would appreciate some comments to explain.

**Review by afd — CHANGES_REQUESTED** [review]

File: coverage/PENDING_REVIEW/intel/0451-COVERAGE-nir-loop-analyze-834-937-995/cov-texel-double-negation.amber (hunk 1 above)
Comment (asuonpaa) [reply coverage/PENDING_REVIEW/intel/0451-COVERAGE-nir-loop-analyze-834-937-995/cov-texel-double-negation.amber:38]: You're looking at the texture generation code provided by gfauto. The manually reduced shader is below that.

Comment (afd) [reply coverage/PENDING_REVIEW/intel/0451-COVERAGE-nir-loop-analyze-834-937-995/cov-texel-double-negation.amber:38]: Haha, thanks for clarifying!

File: coverage/PENDING_REVIEW/intel/0451-COVERAGE-nir-loop-analyze-834-937-995/cov-texel-double-negation.amber [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+#
+# layout(set = 0, binding = 2) uniform sampler2D tex;
+#
+# void main()
+# {
+#     int i = 1;
+#     vec4 texel = vec4(_float_1_0);
+#
+#     // Iterates until i becomes 5.
+#     while(!(false || (!(texel.z != _float_2_0 && i < 5))))
```
Comment (afd) [inline coverage/PENDING_REVIEW/intel/0451-COVERAGE-nir-loop-analyze-834-937-995/cov-texel-double-negation.amber:87]: It would be good to say something about what properties `texel` is guaranteed to have (due to the auto-generated texgen shader), so that the reader of this code feels reassured that loop termination will only depend on i.

**Review by afd — CHANGES_REQUESTED** [review]

**Comment by Vihanakangas** [comment]
@afd Updated.

**Review by afd — APPROVED** [review]
